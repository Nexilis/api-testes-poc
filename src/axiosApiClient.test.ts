/// <reference types="jest" />
/// <reference types="axios" />

import Axios from "axios";

test("Axios api test", async () => {
    var url = new URL("https://jsonvat.com/");

    var response = await Axios.get(url.href);

    expect(response.status).toBe(200);
    expect(response.data).toMatchSnapshot();
});