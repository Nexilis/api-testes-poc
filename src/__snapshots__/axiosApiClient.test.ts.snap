// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Axios api test 1`] = `
Object {
  "details": "http://github.com/adamcooke/vat-rates",
  "rates": Array [
    Object {
      "code": "ES",
      "country_code": "ES",
      "name": "Spain",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 10,
            "standard": 21,
            "super_reduced": 4,
          },
        },
      ],
    },
    Object {
      "code": "BG",
      "country_code": "BG",
      "name": "Bulgaria",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 9,
            "standard": 20,
          },
        },
      ],
    },
    Object {
      "code": "HU",
      "country_code": "HU",
      "name": "Hungary",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 18,
            "standard": 27,
          },
        },
      ],
    },
    Object {
      "code": "LV",
      "country_code": "LV",
      "name": "Latvia",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 12,
            "standard": 21,
          },
        },
      ],
    },
    Object {
      "code": "PL",
      "country_code": "PL",
      "name": "Poland",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 8,
            "standard": 23,
          },
        },
      ],
    },
    Object {
      "code": "UK",
      "country_code": "GB",
      "name": "United Kingdom",
      "periods": Array [
        Object {
          "effective_from": "2011-01-04",
          "rates": Object {
            "reduced": 5,
            "standard": 20,
          },
        },
      ],
    },
    Object {
      "code": "CZ",
      "country_code": "CZ",
      "name": "Czech Republic",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 15,
            "standard": 21,
          },
        },
      ],
    },
    Object {
      "code": "MT",
      "country_code": "MT",
      "name": "Malta",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 7,
            "standard": 18,
          },
        },
      ],
    },
    Object {
      "code": "IT",
      "country_code": "IT",
      "name": "Italy",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 10,
            "standard": 22,
            "super_reduced": 4,
          },
        },
      ],
    },
    Object {
      "code": "SI",
      "country_code": "SI",
      "name": "Slovenia",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 9.5,
            "standard": 22,
          },
        },
      ],
    },
    Object {
      "code": "IE",
      "country_code": "IE",
      "name": "Ireland",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "parking": 13.5,
            "reduced1": 9,
            "reduced2": 13.5,
            "standard": 23,
            "super_reduced": 4.8,
          },
        },
      ],
    },
    Object {
      "code": "SE",
      "country_code": "SE",
      "name": "Sweden",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 6,
            "reduced2": 12,
            "standard": 25,
          },
        },
      ],
    },
    Object {
      "code": "DK",
      "country_code": "DK",
      "name": "Denmark",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "standard": 25,
          },
        },
      ],
    },
    Object {
      "code": "FI",
      "country_code": "FI",
      "name": "Finland",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 10,
            "reduced2": 14,
            "standard": 24,
          },
        },
      ],
    },
    Object {
      "code": "CY",
      "country_code": "CY",
      "name": "Cyprus",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 9,
            "standard": 19,
          },
        },
      ],
    },
    Object {
      "code": "LU",
      "country_code": "LU",
      "name": "Luxembourg",
      "periods": Array [
        Object {
          "effective_from": "2016-01-01",
          "rates": Object {
            "parking": 13,
            "reduced1": 8,
            "standard": 17,
            "super_reduced": 3,
          },
        },
        Object {
          "effective_from": "2015-01-01",
          "rates": Object {
            "parking": 12,
            "reduced1": 8,
            "reduced2": 14,
            "standard": 17,
            "super_reduced": 3,
          },
        },
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "parking": 12,
            "reduced1": 6,
            "reduced2": 12,
            "standard": 15,
            "super_reduced": 3,
          },
        },
      ],
    },
    Object {
      "code": "RO",
      "country_code": "RO",
      "name": "Romania",
      "periods": Array [
        Object {
          "effective_from": "2017-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 9,
            "standard": 19,
          },
        },
        Object {
          "effective_from": "2016-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 9,
            "standard": 20,
          },
        },
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 9,
            "standard": 24,
          },
        },
      ],
    },
    Object {
      "code": "EE",
      "country_code": "EE",
      "name": "Estonia",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 9,
            "standard": 20,
          },
        },
      ],
    },
    Object {
      "code": "EL",
      "country_code": "GR",
      "name": "Greece",
      "periods": Array [
        Object {
          "effective_from": "2016-06-01",
          "rates": Object {
            "reduced1": 6,
            "reduced2": 13.5,
            "standard": 24,
          },
        },
        Object {
          "effective_from": "2016-01-01",
          "rates": Object {
            "reduced1": 6,
            "reduced2": 13.5,
            "standard": 23,
          },
        },
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 6.5,
            "reduced2": 13,
            "standard": 23,
          },
        },
      ],
    },
    Object {
      "code": "LT",
      "country_code": "LT",
      "name": "Lithuania",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 9,
            "standard": 21,
          },
        },
      ],
    },
    Object {
      "code": "FR",
      "country_code": "FR",
      "name": "France",
      "periods": Array [
        Object {
          "effective_from": "2014-01-01",
          "rates": Object {
            "reduced1": 5.5,
            "reduced2": 10,
            "standard": 20,
            "super_reduced": 2.1,
          },
        },
        Object {
          "effective_from": "2012-01-01",
          "rates": Object {
            "reduced1": 5.5,
            "reduced2": 7,
            "standard": 19.6,
            "super_reduced": 2.1,
          },
        },
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5.5,
            "standard": 19.6,
            "super_reduced": 2.1,
          },
        },
      ],
    },
    Object {
      "code": "HR",
      "country_code": "HR",
      "name": "Croatia",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced1": 5,
            "reduced2": 13,
            "standard": 25,
          },
        },
      ],
    },
    Object {
      "code": "BE",
      "country_code": "BE",
      "name": "Belgium",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "parking": 12,
            "reduced1": 6,
            "reduced2": 12,
            "standard": 21,
          },
        },
      ],
    },
    Object {
      "code": "NL",
      "country_code": "NL",
      "name": "Netherlands",
      "periods": Array [
        Object {
          "effective_from": "2012-10-01",
          "rates": Object {
            "reduced": 6,
            "standard": 21,
          },
        },
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 6,
            "standard": 19,
          },
        },
      ],
    },
    Object {
      "code": "SK",
      "country_code": "SK",
      "name": "Slovakia",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 10,
            "standard": 20,
          },
        },
      ],
    },
    Object {
      "code": "DE",
      "country_code": "DE",
      "name": "Germany",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "reduced": 7,
            "standard": 19,
          },
        },
      ],
    },
    Object {
      "code": "PT",
      "country_code": "PT",
      "name": "Portugal",
      "periods": Array [
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "parking": 13,
            "reduced1": 6,
            "reduced2": 13,
            "standard": 23,
          },
        },
      ],
    },
    Object {
      "code": "AT",
      "country_code": "AT",
      "name": "Austria",
      "periods": Array [
        Object {
          "effective_from": "2016-01-01",
          "rates": Object {
            "parking": 13,
            "reduced1": 10,
            "reduced2": 13,
            "standard": 20,
          },
        },
        Object {
          "effective_from": "0000-01-01",
          "rates": Object {
            "parking": 12,
            "reduced": 10,
            "standard": 20,
          },
        },
      ],
    },
  ],
  "version": null,
}
`;
